optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/GizmoMemberAccessorCreator.java

```diff
@@ -0,0 +1,265 @@
+
+        GizmoMemberAccessor.MemberDescriptor member;
+        String name;
+        MethodInfo setterMethod;
+
+        if (methodInfo.name().startsWith("get")) {
+            name = methodInfo.name().substring(3, 4).toLowerCase(Locale.ROOT) + methodInfo.name().substring(4);
+            setterMethod = classInfo.method("set" + name.substring(0, 1).toUpperCase(Locale.ROOT) + name.substring(1),
+                    methodInfo.returnType());
+        } else if (methodInfo.name().startsWith("is")) {
+            name = methodInfo.name().substring(2, 3).toLowerCase(Locale.ROOT) + methodInfo.name().substring(3);
+            setterMethod = classInfo.method("set" + name.substring(0, 1).toUpperCase(Locale.ROOT) + name.substring(1),
+                    methodInfo.returnType());
+        } else {
+            name = methodInfo.name();
+            setterMethod = null;
+        }
+
+        Optional<MethodDescriptor> setterDescriptor = Optional.ofNullable(setterMethod).map(MethodDescriptor::of);
+        if (Modifier.isPublic(methodInfo.flags())) {
+            member = new GizmoMemberAccessor.MemberDescriptor(methodInfo, methodInfo, setterDescriptor, indexView, name);
+        } else {
+            setterDescriptor = addVirtualMethodGetter(classInfo, methodInfo, name, setterDescriptor, transformers);
+            String methodName = getVirtualGetterName(name);
+            MethodInfo newMethodInfo = classInfo.method(methodName);
+            member = new GizmoMemberAccessor.MemberDescriptor(methodInfo, newMethodInfo, setterDescriptor, indexView, name);
+        }
+        try {
+            GizmoMemberAccessor.defineAccessorFor(classCreator, member,
+                    (Class<? extends Annotation>) Class.forName(annotationInstance.name().toString()));
+            classCreator.close();
+        } catch (ClassNotFoundException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private static class OptaPlannerFieldEnhancingClassVisitor extends ClassVisitor {
+        private final FieldInfo fieldInfo;
+        private final Class<?> clazz;
+        private final String fieldTypeDescriptor;
+
+        public OptaPlannerFieldEnhancingClassVisitor(ClassInfo classInfo, ClassVisitor outputClassVisitor,
+                FieldInfo fieldInfo) {
+            super(Gizmo.ASM_API_VERSION, outputClassVisitor);
+            this.fieldInfo = fieldInfo;
+            try {
+                clazz = Class.forName(classInfo.name().toString(), false, Thread.currentThread().getContextClassLoader());
+                fieldTypeDescriptor = DescriptorUtils.typeToString(fieldInfo.type());
+            } catch (ClassNotFoundException e) {
+                throw new IllegalStateException(e);
+            }
+        }
+
+        @Override
+        public void visitEnd() {
+            super.visitEnd();
+            addGetter(this.cv);
+            addSetter(this.cv);
+        }
+
+        private void addSetter(ClassVisitor classWriter) {
```

Comment:
Why is it necessary to use directly ASM?

Reply:
Cannot use Gizmo for argumentation (modifying existing classes, which we do so we have a public getter/setter if their field/method is private)